FinniversKit/Sources/Base Components/ButtonStyle.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Copyright © 2020 FINN AS. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13.0, *)
+public struct InlineFlatStyle: ButtonStyle {
+    private let font: Font
+    private let textColor: Color
+
+    public init(size: Button.Size = .normal, textColor: Color = .btnPrimary) {
+        self.font = size == .normal ? .finnFont(.bodyStrong) : .finnFont(.detailStrong)
+        self.textColor = textColor
+    }
+
+    public func makeBody(configuration: Configuration) -> some View {
+        configuration
+            .label
+            .font(font)
+            .foregroundColor(configuration.isPressed ? textColor.opacity(0.8) : textColor)
+    }
+}
+
+@available(iOS 13.0, *)
+public struct FlatStyle: ButtonStyle {
+    private let size: Button.Size
+    private let font: Font
+    private let textColor: Color
+
+    public init(size: Button.Size = .normal, textColor: Color = .btnPrimary) {
+        self.size = size
+        self.font = size == .normal ? .finnFont(.bodyStrong) : .finnFont(.detailStrong)
+        self.textColor = textColor
+    }
+
+    public func makeBody(configuration: Configuration) -> some View {
+        HStack {
+            Spacer()
+            InlineFlatStyle(size: size, textColor: textColor)
+                .makeBody(configuration: configuration)
+            Spacer()
+        }
+        .padding(.vertical, .spacingS)
+        .padding(.horizontal, .spacingM)
+    }
+}
+
+@available(iOS 13.0, *)
+public struct DefaultStyle: ButtonStyle {
+    private let size: Button.Size
+    private let font: Font
+    private let textColor: Color
+
+    public init(size: Button.Size = .normal, textColor: Color = .btnPrimary) {
+        self.size = size
+        self.font = size == .normal ? .finnFont(.bodyStrong) : .finnFont(.detailStrong)
+        self.textColor = textColor
+    }
+
+    public func makeBody(configuration: Configuration) -> some View {
+        HStack {
+            Spacer()
+            configuration
+                .label
+                .font(font)
+                .foregroundColor(textColor)
+            Spacer()
+        }
+        .padding(.vertical, .spacingS)
+        .padding(.horizontal, .spacingM)
+        .background(
+            configuration.isPressed ? Color(.defaultButtonHighlightedBodyColor) : Color.bgPrimary
```

Comment:
Have you tested this in dark mode? I got some weird blinking with using `defaultButtonHighlightedBodyColor` on press in basic list cells.

Reply:
yes, it works well